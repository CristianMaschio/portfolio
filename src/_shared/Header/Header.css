header {
  width: 18rem;
  height: 100%;
  display: flex;
  position: fixed;
  border-right-style: solid;
  border-right-width: 0.5px;
  border-right-color: white;
}

.header-blue {
  width: 1em;
  background-color: #3b62b1;
  height: 100%;
}

.header-information {
  width: 100%;
  height: 100%;
  color: white;
  background-color: #313131;
}

.image-profile {
  width: 200px;
  height: 200px;
  border-radius: 50%;
}

.image-container {
  border-radius: 50%;
  border: 10px;
  width: 210px;
  height: 210px;
  background-image: linear-gradient(to right, #3b62b1, transparent);
  border-color: black;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: auto;
  margin-top: 1em;
}

.menuContainer {
  color: white;
  margin-top: 1rem;
}

.menuContainer h2 {
  margin: 0.8rem;
  margin-left: 1rem;
  font-weight: 500;
}

.menuContent {
  margin-left: 1rem;
}

.menuContentContainer {
  display: flex;
  flex-direction: row;
}

.menuContentContainer p {
  color: white;
  flex: 1;
  margin: 0.5rem;
  margin-left: 1rem;
}

.menuContent a {
  color: white;
  margin: 1rem;
  font-size: 1.5rem;
  flex: 1;
}

.line {
  background-image: linear-gradient(to right, #3b62b1, transparent);
  height: 1px;
}
