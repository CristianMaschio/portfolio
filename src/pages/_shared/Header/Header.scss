@import "../../../styles/colors";

header {
  width: 18rem;
  height: 100%;
  display: flex;
  position: fixed;
  border-right-style: solid;
  border-right-width: 0.5px;
  border-right-color: white;
  background-color: #3b62b1;
  @media screen and (max-width: $mobile-max-width) {
    position: sticky;
    top: 0;
    width: 100vw;
    height: 100vh;
    z-index: 2;
  }

  .header-blue {
    width: 1em;
    background-color: #3b62b1;
    height: 100%;
    @media screen and (max-width: $mobile-max-width) {
      height: 100vh;
    }
  }

  .header-information {
    width: 100%;
    height: 100%;
    color: white;
    background-color: #313131;
    @media screen and (max-width: $mobile-max-width) {
      max-width: 18rem;
    }
  }

  .menuContainer {
    color: white;
    margin-top: 1rem;
  }

  .menuContainer h2 {
    margin: 0.8rem;
    margin-left: 1rem;
    font-weight: 500;
  }

  .menuContent {
    margin-left: 1rem;
  }

  .menuContentContainer {
    display: flex;
    flex-direction: row;
  }

  .menuContentContainer p {
    color: white;
    flex: 1;
    margin: 0.5rem;
    margin-left: 1rem;
  }

  .menuContent a {
    color: white;
    margin: 1rem;
    font-size: 1.5rem;
    flex: 1;
  }

  .line {
    background-image: $primary-right-linear-gradient;
    height: 1px;
  }

  @media screen and (max-width: $mobile-max-width) {
    .close {
      position: absolute;
      right: 0;
      top: 0;
      width: 32px;
      height: 32px;
      padding: 15px;
      opacity: 0.3;
    }
    .close:hover {
      opacity: 1;
    }
    .close:before,
    .close:after {
      position: absolute;
      left: 30px;
      content: " ";
      height: 33px;
      width: 3px;
      background-color: white;
    }
    .close:before {
      transform: rotate(45deg);
    }
    .close:after {
      transform: rotate(-45deg);
    }
  }
}

.only-desktop {
  display: none;
  @media screen and (min-width: $mobile-max-width) {
    display: flex;
  }
}
