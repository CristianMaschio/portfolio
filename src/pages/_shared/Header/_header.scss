.open-header {
  margin-left: 0px;
  opacity: 100;
}

header {
  width: 18rem;
  height: 100vh;
  display: flex;
  position: fixed;
  border-right-style: solid;
  border-right-width: 0.5px;
  border-right-color: white;
  background-color: $background-primary-color;
  @media screen and (max-width: $mobile-max-width) {
    position: fixed;
    top: 0;
    opacity: 0;
    width: 100vw;
    max-height: 100vh;
    height: 100%;
    z-index: 10;
    margin-left: -100vw;
    transition: all 0.4s;
    overflow-y: scroll;
    background-color: rgba(20, 35, 71, 0.9);
  }

  .header-blue {
    width: 0.5em;
    background-color: $primary-color;
    height: 100%;
    @media screen and (max-width: $mobile-max-width) {
      height: 100vh;
    }
  }

  .header-information {
    width: 100%;
    color: black;
    background-color: rgba(255, 255, 255, 0.9);
    @media screen and (max-width: $mobile-max-width) {
      position: relative;
      z-index: 11;
      max-width: 18rem;
      height: 100%;
      overflow: auto;
    }
  }

  .header-container {
    margin-top: 1rem;
    h2 {
      margin: 0.8rem;
      margin-left: 1rem;
      font-weight: 500;
    }
  }

  .header-content {
    margin-left: 1rem;
  }

  a {
    color: $primary-color;
    margin: 1rem 1rem 0 1rem;
    font-size: 1.5rem;
    flex: 1;
  }

  .row-content {
    display: flex;
    flex-direction: row;
    p {
      flex: 1;
      margin: 0.5rem;
      margin-left: 1rem;
    }
  }

  .download-cv {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    .title {
      flex: 1;
    }
    a {
      margin: 0 1rem;
      flex: 0;
    }
  }

  .line {
    background-image: $primary-right-linear-gradient;
    height: 0.8px;
  }

  @media screen and (max-width: $mobile-max-width) {
    .close {
      position: absolute;
      z-index: 1000;
      right: 0;
      top: 0;
      width: 32px;
      height: 32px;
      padding: 15px;
      opacity: 0.7;
    }
    .close:hover {
      opacity: 1;
      cursor: pointer;
    }
    .close:before,
    .close:after {
      position: absolute;
      left: 30px;
      content: " ";
      height: 33px;
      width: 3px;
      background-color: black;
    }
    .close:before {
      transform: rotate(45deg);
    }
    .close:after {
      transform: rotate(-45deg);
    }
  }
  .eu-regolation {
    font-size: 10px;
    margin-right: 10%;
    margin-left: 10%;
    padding-bottom: 1rem;
    text-align: center;
    margin-top: 1rem;
  }
}

.only-desktop {
  display: none;
  @media screen and (min-width: $mobile-max-width) {
    display: flex;
  }
}
