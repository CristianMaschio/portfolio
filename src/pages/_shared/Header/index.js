import React, { useContext } from "react"
import Icon from "antd/lib/icon"

import ImageProfile from "../ImageProfile"
import AppState from "../../../gState/context"

function HeaderContent(props) {
  return (
    <div>
      {props.personalData.map(content => {
        return (
          <div key={content.key} className="header-content">
            <div className="line" />
            <div className="row-content">
              <p>{content.key}:</p>
              <p>{content.text}</p>
            </div>
          </div>
        )
      })}
    </div>
  )
}

function HeaderContacts() {
  return (
    <div className="header-content">
      <div className="line" />
      <div className="row-content">
        <a href="tel:345-273-6368" target="_blank" rel="noopener noreferrer">
          <Icon type="phone" />
        </a>
        <a
          href="mailto:cristianmaschio@hotmail.it"
          target="_blank"
          rel="noopener noreferrer"
        >
          <Icon type="mail" />
        </a>
        <a
          href="https://github.com/CristianMaschio"
          target="_blank"
          rel="noopener noreferrer"
        >
          <Icon type="github" />
        </a>
        <a
          href="https://www.linkedin.com/in/cristian-maschio-37004a146"
          target="_blank"
          rel="noopener noreferrer"
        >
          <Icon type="linkedin" />
        </a>
      </div>
    </div>
  )
}

export default function Header() {
  const { headerClass, setRenderInformation } = useContext(AppState)

  const headerContent = require(`../../../assets/languages/En-content.json`)

  const closeHeaderInformation = event => {
    event.stopPropagation()
    setRenderInformation(false)
  }

  const stopPropagation = event => {
    event.stopPropagation()
  }

  return (
    <header className={headerClass} onClick={closeHeaderInformation}>
      <div className="header-blue" />
      <div className="header-information" onClick={stopPropagation}>
        <div onClick={closeHeaderInformation} className="close" />
        <ImageProfile size={200} />
        <div className="header-container">
          <div className="line" />
          <h2>{headerContent.header.personal}</h2>
          <HeaderContent personalData={headerContent.header.personalData} />
        </div>
        <div className="header-container">
          <div className="line" />
          <h2>{headerContent.header.contacts}</h2>
          <HeaderContacts />
        </div>
        <div className="header-container">
          <div className="line" />
          <h2>
            <div className="download-cv">
              <div className="title">Download CV</div>
              <a
                href="https://drive.google.com/uc?export=download&id=1lB-ySL7IN_8K1UNJVK8TWG9OqxAvVtdG"
                target="_blank"
                rel="noopener noreferrer"
              >
                <Icon type="download" />
              </a>
            </div>
          </h2>
          <div className="header-content">
            <div className="line" />
            <div className="row-content"></div>
          </div>
        </div>
        <div className="eu-regolation">{headerContent.privacyTerms}</div>
      </div>
    </header>
  )
}
