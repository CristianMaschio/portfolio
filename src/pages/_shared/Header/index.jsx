import React, { useContext } from "react"
import {
  LinkedinFilled,
  MailFilled,
  GithubFilled,
  CloudDownloadOutlined,
} from "@ant-design/icons"

import ImageProfile from "../ImageProfile"
import Context from "../../../Context"

function HeaderContent(props) {
  return (
    <div>
      {props.personalData.map((content) => {
        return (
          <div key={content.key} className="header-content">
            <div className="line" />
            <div className="row-content">
              <p>{content.key}:</p>
              <p>{content.text}</p>
            </div>
          </div>
        )
      })}
    </div>
  )
}

function HeaderContacts() {
  return (
    <div className="header-content">
      <div className="line" />
      <div className="row-content">
        <a
          href="mailto:cristianmaschio@hotmail.it"
          target="_blank"
          rel="noopener noreferrer"
        >
          <MailFilled />
        </a>
        <a
          href="https://github.com/CristianMaschio"
          target="_blank"
          rel="noopener noreferrer"
        >
          <GithubFilled />
        </a>
        <a
          href="https://www.linkedin.com/in/cristian-maschio-37004a146"
          target="_blank"
          rel="noopener noreferrer"
        >
          <LinkedinFilled />
        </a>
      </div>
    </div>
  )
}

export default function Header() {
  const { headerClass, setRenderInformation } = useContext(Context)

  const headerContent = require(`../../../assets/languages/En-content.json`)

  const closeHeaderInformation = (event) => {
    event.stopPropagation()
    setRenderInformation(false)
  }

  const stopPropagation = (event) => {
    event.stopPropagation()
  }

  return (
    <header className={headerClass} onClick={closeHeaderInformation}>
      <div className="header-blue" />
      <div className="header-information" onClick={stopPropagation}>
        <div onClick={closeHeaderInformation} className="close" />
        <ImageProfile size={200} />
        <div className="header-container">
          <div className="line" />
          <h2>{headerContent.header.personal}</h2>
          <HeaderContent personalData={headerContent.header.personalData} />
        </div>
        <div className="header-container">
          <div className="line" />
          <h2>{headerContent.header.contacts}</h2>
          <HeaderContacts />
        </div>
        <div className="header-container">
          <div className="line" />
          <h2>
            <div className="download-cv">
              <div className="title">Download CV</div>
              <a
                href="https://drive.google.com/uc?export=download&id=1lB-ySL7IN_8K1UNJVK8TWG9OqxAvVtdG"
                target="_blank"
                rel="noopener noreferrer"
              >
                <CloudDownloadOutlined />
              </a>
            </div>
          </h2>
          <div className="header-content">
            <div className="line" />
            <div className="row-content" />
          </div>
        </div>
        <div className="eu-regolation">{headerContent.privacyTerms}</div>
      </div>
    </header>
  )
}
