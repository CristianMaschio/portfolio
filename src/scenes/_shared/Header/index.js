import "./Header.scss"
import React, { useContext } from "react"
import Icon from "antd/lib/icon"

import ImageProfile from "../ImageProfile"
import AppState from "../../../gState/context"

function HeaderContent(props) {
  return (
    <div>
      {props.personalData.map(content => {
        return (
          <div key={content.key} className="header-content">
            <div className="line" />
            <div className="row-content">
              <p>{content.key}:</p>
              <p>{content.text}</p>
            </div>
          </div>
        )
      })}
    </div>
  )
}

function HeaderContacts() {
  return (
    <div className="header-content">
      <div className="line" />
      <div className="row-content">
        <a href="tel:345-273-6368" target="_blank" rel="noopener noreferrer">
          <Icon type="phone" />
        </a>
        <a
          href="mailto:cristianmaschio@hotmail.it"
          target="_blank"
          rel="noopener noreferrer"
        >
          <Icon type="mail" />
        </a>
        <a
          href="https://github.com/CristianMaschio"
          target="_blank"
          rel="noopener noreferrer"
        >
          <Icon type="github" />
        </a>
        <a
          href="https://www.linkedin.com/in/cristian-maschio-37004a146"
          target="_blank"
          rel="noopener noreferrer"
        >
          <Icon type="linkedin" />
        </a>
      </div>
    </div>
  )
}

export default function Header() {
  const { headerClass, setRenderInformation, language } = useContext(AppState)

  const headerContent = require(`../../../data/${language}-content.json`)

  const closeHeaderInformation = event => {
    event.stopPropagation()
    setRenderInformation(false)
  }

  const stopPropagation = event => {
    event.stopPropagation()
  }

  return (
    <header className={headerClass} onClick={closeHeaderInformation}>
      <div className="header-blue" />
      <div className="header-information" onClick={stopPropagation}>
        <div onClick={closeHeaderInformation} className="close" />
        <ImageProfile size={200} />
        <div className="header-container">
          <div className="line" />
          <h2>{headerContent.header.personal}</h2>
          <HeaderContent personalData={headerContent.header.personalData} />
        </div>
        <div className="header-container">
          <div className="line" />
          <h2>{headerContent.header.contacts}</h2>
          <HeaderContacts />
        </div>
        <div className="eu-regolation">{headerContent.privacyTerms}</div>
      </div>
    </header>
  )
}
