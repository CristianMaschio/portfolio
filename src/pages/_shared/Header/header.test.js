import React from "react"
import Header from "../Header"
import renderer from "react-test-renderer"

it("Header: render correctly", () => {
  const header = renderer.create(<Header />).toJSON()
  expect(header).toMatchSnapshot()
})
