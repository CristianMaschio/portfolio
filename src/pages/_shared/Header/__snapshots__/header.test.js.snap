// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header: render correctly 1`] = `
<header
  className=""
  onClick={[Function]}
>
  <div
    className="header-blue"
  />
  <div
    className="header-information"
    onClick={[Function]}
  >
    <div
      className="close"
      onClick={[Function]}
    />
    <div
      className="image-container"
    >
      <div
        className="image-background-animation"
        style={
          Object {
            "height": 210,
            "width": 210,
          }
        }
      />
      <img
        alt="profile"
        className="image-profile"
        src="imageProfile.jpg"
        style={
          Object {
            "height": 200,
            "width": 200,
          }
        }
      />
    </div>
    <div
      className="header-container"
    >
      <div
        className="line"
      />
      <h2>
        Personal
      </h2>
      <div>
        <div
          className="header-content"
        >
          <div
            className="line"
          />
          <div
            className="row-content"
          >
            <p>
              Name
              :
            </p>
            <p>
              Cristian
            </p>
          </div>
        </div>
        <div
          className="header-content"
        >
          <div
            className="line"
          />
          <div
            className="row-content"
          >
            <p>
              Surname
              :
            </p>
            <p>
              Maschio
            </p>
          </div>
        </div>
        <div
          className="header-content"
        >
          <div
            className="line"
          />
          <div
            className="row-content"
          >
            <p>
              Birthday
              :
            </p>
            <p>
              31/05/1994
            </p>
          </div>
        </div>
        <div
          className="header-content"
        >
          <div
            className="line"
          />
          <div
            className="row-content"
          >
            <p>
              Location
              :
            </p>
            <p>
              North London
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="header-container"
    >
      <div
        className="line"
      />
      <h2>
        Contacts
      </h2>
      <div
        className="header-content"
      >
        <div
          className="line"
        />
        <div
          className="row-content"
        >
          <a
            href="mailto:cristianmaschio@hotmail.it"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              aria-label="mail"
              className="anticon anticon-mail"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="mail"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-80.8 108.9L531.7 514.4c-7.8 6.1-18.7 6.1-26.5 0L189.6 268.9A7.2 7.2 0 01194 256h648.8a7.2 7.2 0 014.4 12.9z"
                />
              </svg>
            </span>
          </a>
          <a
            href="https://github.com/CristianMaschio"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              aria-label="github"
              className="anticon anticon-github"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="github"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                />
              </svg>
            </span>
          </a>
          <a
            href="https://www.linkedin.com/in/cristian-maschio-37004a146"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              aria-label="linkedin"
              className="anticon anticon-linkedin"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="linkedin"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1168.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
                />
              </svg>
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      className="header-container"
    >
      <div
        className="line"
      />
      <h2>
        <div
          className="download-cv"
        >
          <div
            className="title"
          >
            Download CV
          </div>
          <a
            href="https://drive.google.com/uc?export=download&id=1lB-ySL7IN_8K1UNJVK8TWG9OqxAvVtdG"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              aria-label="cloud-download"
              className="anticon anticon-cloud-download"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="cloud-download"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M624 706.3h-74.1V464c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v242.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.7a8 8 0 0012.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9z"
                />
                <path
                  d="M811.4 366.7C765.6 245.9 648.9 160 512.2 160S258.8 245.8 213 366.6C127.3 389.1 64 467.2 64 560c0 110.5 89.5 200 199.9 200H304c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8h-40.1c-33.7 0-65.4-13.4-89-37.7-23.5-24.2-36-56.8-34.9-90.6.9-26.4 9.9-51.2 26.2-72.1 16.7-21.3 40.1-36.8 66.1-43.7l37.9-9.9 13.9-36.6c8.6-22.8 20.6-44.1 35.7-63.4a245.6 245.6 0 0152.4-49.9c41.1-28.9 89.5-44.2 140-44.2s98.9 15.3 140 44.2c19.9 14 37.5 30.8 52.4 49.9 15.1 19.3 27.1 40.7 35.7 63.4l13.8 36.5 37.8 10C846.1 454.5 884 503.8 884 560c0 33.1-12.9 64.3-36.3 87.7a123.07 123.07 0 01-87.6 36.3H720c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h40.1C870.5 760 960 670.5 960 560c0-92.7-63.1-170.7-148.6-193.3z"
                />
              </svg>
            </span>
          </a>
        </div>
      </h2>
      <div
        className="header-content"
      >
        <div
          className="line"
        />
        <div
          className="row-content"
        />
      </div>
    </div>
    <div
      className="eu-regolation"
    >
      I authorize the processing of my personal data pursuant to Legislative Decree 196/03 and the General Data Protection Regulation (EU / 2016/679).
    </div>
  </div>
</header>
`;
